File: MATTHEW/leetcode/AIcoding/2.Gemini-2.0/java_solutions/3046-split-the-array.java
class Solution {
    public boolean splitArray(int[] nums) {
        int n = nums.length;
        if (n < 7) {
            return false;
        }

        int[] sum = new int[n];
        sum[0] = nums[0];
        for (int i = 1; i < n; i++) {
            sum[i] = sum[i - 1] + nums[i];
        }

        for (int j = 3; j < n - 3; j++) {
            for (int i = 1; i < j - 1; i++) {
                if (sum[i - 1] == sum[j - 1] - sum[i]) {
                    for (int k = j + 2; k < n - 1; k++) {
                        if (sum[k - 1] - sum[j] == sum[n - 1] - sum[k] &&
                            sum[i - 1] == sum[k - 1] - sum[j]) {
                            return true;
                        }
                    }
                }
            }
        }

        return false;
    }
}

File: leetcode/AIcoding/2.Gemini-2.0/rust_solutions/3046-split-the-array.rs
impl Solution {
    pub fn split_array(nums: Vec<i32>) -> bool {
        let n = nums.len();
        if n < 7 {
            return false;
        }

        let mut sum = vec![0; n + 1];
        for i in 0..n {
            sum[i + 1] = sum[i] + nums[i];
        }

        for j in 3..n - 3 {
            let sum_j = sum[j];
            let mut i_set = std::collections::HashSet::new();
            for i in 1..j - 1 {
                if sum[i] == sum_j - sum[i + 1] {
                    i_set.insert(i);
                }
            }

            let sum_j_1 = sum[j + 1];
            for k in j + 2..n - 1 {
                if sum[k] - sum_j_1 == sum[n] - sum[k + 1] {
                    for &i in &i_set {
                        if sum[i] == sum[k] - sum_j_1 {
                            return true;
                        }
                    }
                }
            }
        }

        false
    }
}